docs/marketplace-and-submit-job-v2/marketplace-and-submit-job-v2.md

```diff
@@ -79,40 +79,280 @@ And you can share the script, data, docker images or whole job configuration on
+| PAI_CURRENT_TASK_ROLE_CPU_COUNT    | `taskRole.cpuNumber` of current task role  |
+| PAI_CURRENT_TASK_ROLE_MEM_MB       | `taskRole.memoryMB` of current task role   |
+| PAI_CURRENT_TASK_ROLE_SHM_MB       | `taskRole.shmMB` of current task role      |
+| PAI_CURRENT_TASK_ROLE_GPU_COUNT    | `taskRole.gpuNumber` of current task role  |
+| PAI_CURRENT_TASK_ROLE_MIN_FAILED_TASK_COUNT    | `taskRole.minFailedTaskCount` of current task role    |
+| PAI_CURRENT_TASK_ROLE_MIN_SUCCEEDED_TASK_COUNT | `taskRole.minSucceededTaskCount` of current task role |
+| PAI_CURRENT_TASK_ROLE_CURRENT_TASK_INDEX | Index of current task in current task role, starting from 0 |
+| PAI_JOB_TASK_COUNT                 | Total tasks' number in config file       |
+| PAI_JOB_TASK_ROLE_COUNT            | Total task roles' number in config file  |
+| PAI_JOB_TASK_ROLE_LIST             | Comma separated all task role names in config file |
+| PAI_CONTAINER_HOST_IP              | Allocated ip for current docker container |
+| PAI_CONTAINER_HOST_PORT_LIST       | Allocated port list for current docker container, in `portLabel0:port0,port1,port2;portLabel1:port3,port4` format |
+| PAI_CONTAINER_HOST\_`$type`\_PORT_LIST | Allocated port list for `portList.label == $type`, comma separated `port` string |
+| PAI_TASK_ROLE\_`$name`\_HOST_LIST  | Host list for `PAI_TASK_ROLE_NAME == $name`, comma separated `ip:port` string, sorted by current task index in task role. Each task role has a host list environment variable with the corresponding task role name |
+
+Below is an example for a distributed tensorflow image classification training job :
 
 ```yaml
 protocol_version: v2
-name : tensorflow_serving_mnist
-type : job
-version : 1.0.0
-contributor : Qi chen
-description : image classification, mnist dataset, tensorflow, serving
-retryCount: -2
-
-tasks :
-  - role : worker
-    dockerimage : tf_serving_example
+name: tensorflow_cifar10              # name of the job
+type: job
+version: 1.0.0
+contributor: Alice                    # contributor of the job
+description: image classification, cifar10 dataset, tensorflow, distributed training
+retryCount: 0                         # job retry count
+virtualCluster: default               # the virtual cluster job runs on
+
+parameters:
+  model: resnet20
+  batchsize: 32
+
+tasks:
+  - role: worker                      # task role name
+    data: cifar10                     # task input data
+    script: tensorflow_cnnbenchmarks  # script executed in task
+    dockerimage: tf_example           # docker image used in task
+    resource:                         # resource required in task 
+      instances: 1                    # number of instances for the task
+      resourcePerInstance: { cpu: 2, memoryMB: 16384, gpu: 4 }
+    minSucceededTaskCount: 1          # number of succeeded tasks to kill the entire job
+    parameters:
+      data: cifar10
+      model: $$model$$
```

Comment:
>$$model$$ [](start = 13, length = 9)

there are multiple "model" in the spec. do we need to introduce a scope. e.g., $$parameters.model$$, $$tasks.parameters.model$$?